### PR TITLE
Add documentation for ipaddr filters 

### PR DESCRIPTION
**Pull request by bmillemathias**
##### SUMMARY
Add documentation for filters network_in_network and network_in_usable
fixes a part of #36388

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME
filter ipaddr

### CODE REVIEW

**Comment by bmillemathias** [comment]
I can't document `reduce_on_network` and `slaac`.
the first one crashed and the latter returned me false on any attempt.

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 1]
```diff
@@ -431,20 +433,70 @@ To find the available range of IP addresses from the given network address, use
...
     192.168.122.0
     # {{ '192.168.122.1/30' | ansible.netcommon.ipaddr('peer') }}
     192.168.122.2
 
+To return the nth ip from a network, use the filter ``nthhost``::
+
+    # {{ '10.0.0.0/8' | ansible.netcommon.nthhost(305) }}
+    10.0.1.49
+
+``nthhost`` supports also a negative value, so it provides the ip address downward::
```
Comment (Akasurde) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
``nthhost`` also supports a negative value::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 2]
```diff
@@ -431,20 +433,70 @@ To find the available range of IP addresses from the given network address, use
...
+
+    # {{ '192.168.122.1/24' | ansible.netcommon.next_nth_usable(2) }}
+    192.168.122.3
+
+If there is no usable address, it returns an empty string::
+
+    # {{ '192.168.122.254/24' | ansible.netcommon.next_nth_usable(2) }}
+    ""
+
+There is also a filter ``previous_nth_usable`` to find previous usable address::
```
Comment (Akasurde) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Just like ``next_nth_ansible``, you have ``previous_nth_usable`` to find previous usable address::
```

Comment (Akasurde) [reply docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]: Just suggestion, not blocker.

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 3]
```diff
@@ -431,20 +433,70 @@ To find the available range of IP addresses from the given network address, use
...
+
+    # {{ '192.168.122.10/24' | ansible.netcommon.previous_nth_usable(2) }}
+    192.168.122.8
+
+Testing if a address belong to a network range
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Filter ``network_in_usable`` returns wether an address passed as argument is usable in a network.
```
Comment (Akasurde) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Filter ``network_in_usable`` returns whether an address passed as argument is usable in a network.
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 4]
```diff
@@ -431,20 +433,70 @@ To find the available range of IP addresses from the given network address, use
...
+    # {{ '192.168.0.0/24' | ansible.netcommon.network_in_usable( '192.168.0.1' ) }}
+    True
+
+    # {{ '192.168.0.0/24' | ansible.netcommon.network_in_usable( '192.168.0.255' ) }}
+    False
+
+    # {{ '192.168.0.0/16' | ansible.netcommon.network_in_usable( '192.168.0.255' ) }}
+    True
+
+Filter ``network_in_network`` returns wether an address or a network passed as argument is in a network.::
```
Comment (Akasurde) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Filter ``network_in_network`` returns whether an address or a network passed as argument is in a network.::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst (hunk 3 above)
Comment (bmillemathias) [reply docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]: fixed

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst (hunk 4 above)
Comment (bmillemathias) [reply docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]: fixed

**Comment by Akasurde** [comment]
LGTM, will wait for @acozine and @samccann to comment.

**Comment by Akasurde** [comment]
@bmillemathias Thanks for the contribution.

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 5]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+
+    # {{ '192.168.122.254/24' | ansible.netcommon.next_nth_usable(2) }}
+    ""
+
+Just like ``next_nth_ansible``, you have ``previous_nth_usable`` to find previous usable address::
+
+    # {{ '192.168.122.10/24' | ansible.netcommon.previous_nth_usable(2) }}
+    192.168.122.8
+
+Testing if a address belong to a network range
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst:475]:
```suggestion
Testing if an address belongs to a network range
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 6]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Filter ``network_in_usable`` returns whether an address passed as argument is usable in a network.
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
The ``network_in_usable`` filter returns whether an address passed as an argument is usable in a network.
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 7]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
-In this example, ``next_nth_usable`` returns the second usable IP address for the given IP range.
-
 To find the peer IP address for a point to point link, use ``peer``::
 
     # {{ '192.168.122.1/31' | ansible.netcommon.ipaddr('peer') }}
     192.168.122.0
     # {{ '192.168.122.1/30' | ansible.netcommon.ipaddr('peer') }}
     192.168.122.2
 
+To return the nth ip from a network, use the filter ``nthhost``::
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst:448]:
```suggestion
To return the nth IP from a network, use the filter ``nthhost``::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 8]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+Just like ``next_nth_ansible``, you have ``previous_nth_usable`` to find previous usable address::
+
+    # {{ '192.168.122.10/24' | ansible.netcommon.previous_nth_usable(2) }}
+    192.168.122.8
+
+Testing if a address belong to a network range
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Filter ``network_in_usable`` returns whether an address passed as argument is usable in a network.
+Usable addresses are addresses that can be assigned to host so address of the network except network
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Usable addresses are addresses that can be assigned to a host. The network ID and
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 9]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+and broadcast addresses.::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
the broadcast address are not usable addresses.::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 10]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+    # {{ '192.168.0.0/24' | ansible.netcommon.network_in_usable( '192.168.0.1' ) }}
+    True
+
+    # {{ '192.168.0.0/24' | ansible.netcommon.network_in_usable( '192.168.0.255' ) }}
+    False
+
+    # {{ '192.168.0.0/16' | ansible.netcommon.network_in_usable( '192.168.0.255' ) }}
+    True
+
+Filter ``network_in_network`` returns whether an address or a network passed as argument is in a network.::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
The ``network_in_network`` filter returns whether an address or a network passed as argument is in a network.::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 11]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+    True
+
+    # {{ '192.168.0.0/24' | ansible.netcommon.network_in_network( '192.168.0.255' ) }}
+    True
+
+    # Check in a network is part of another network
+    # {{ '192.168.0.0/16' | ansible.netcommon.network_in_network( '192.168.0.0/24' ) }}
+    True
+
+Should you have several addresses to check if they belong to a network, you can use ``reduce_on_network``::
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
To check whether multiple addresses belong to a network, use the ``reduce_on_network`` filter::
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 12]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
 
 Here are a few simple examples::
 
+    # Get the next five address based on a ip address
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
    # Get the next five addresses based on an IP address
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 13]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
     # {{ '192.168.1.5' | ansible.netcommon.ipmath(5) }}
     192.168.1.10
 
+    # Get the ten previous address based on a ip address
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
    # Get the ten previous addresses based on an IP address
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 14]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
     # {{ '192.168.0.5' | ansible.netcommon.ipmath(-10) }}
     192.167.255.251
 
+    # Get the next five address using cidr notation
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
    # Get the next five addresses using CIDR notation
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 15]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
     # {{ '192.168.1.1/24' | ansible.netcommon.ipmath(5) }}
     192.168.1.6
 
+    # Get the previous five address using cidr notation
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
    # Get the previous five addresses using CIDR notation
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 16]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
     # {{ '192.168.1.6/24' | ansible.netcommon.ipmath(-5) }}
     192.168.1.1
 
+    # Get the previous ten address using cidr notation
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst:537]:
```suggestion
    # Get the previous ten addresses using CIDR notation
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 17]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
+    # It returns a address of the previous network range
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst:538]:
```suggestion
    # This filter returns an address in the previous network range
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 18]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
     # {{ '192.168.2.6/24' | ansible.netcommon.ipmath(-10) }}
     192.168.1.252
 
+    # Get the next ten address in IPv6
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
    # Get the next ten addresses in IPv6
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 19]
```diff
@@ -455,24 +512,32 @@ The ``ipmath()`` filter can be used to do simple IP math/arithmetic.
...
     # {{ '2001::1' | ansible.netcommon.ipmath(10) }}
     2001::b
 
+    # Get the previous ten address in IPv6
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst:546]:
```suggestion
    # Get the previous ten addresses in IPv6
```

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 20]
```diff
@@ -637,11 +702,20 @@ The supported formats result in the following conversions for the ``1a:2b:3c:4d:
     linux or unix: 1a:2b:3c:4d:5e:6f:
     pgsql, postgresql, or psql: 1a2b3c:4d5e6f
 
+
+Generate IPv6 address in Stateless Configuration (SLAAC)
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Generate an IPv6 address in Stateless Configuration (SLAAC)
```

**Review by acozine — COMMENTED** [review]
Thanks @bmillemathias for a great set of additions. I made some suggestions to make this PR even better.

File: docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst [hunk 21]
```diff
@@ -431,20 +433,75 @@ To find the available range of IP addresses from the given network address, use
...
+
+    # {{ '192.168.122.1/24' | ansible.netcommon.next_nth_usable(2) }}
+    192.168.122.3
+
+If there is no usable address, it returns an empty string::
+
+    # {{ '192.168.122.254/24' | ansible.netcommon.next_nth_usable(2) }}
+    ""
+
+Just like ``next_nth_ansible``, you have ``previous_nth_usable`` to find previous usable address::
```
Comment (samccann) [inline docs/docsite/rst/user_guide/playbooks_filters_ipaddr.rst]:
```suggestion
Just like ``next_nth_ansible``, you have ``previous_nth_usable`` to find the previous usable address::
```

**Review by samccann — APPROVED** [review]

**Comment by Akasurde** [comment]
@bmillemathias Thanks for the contribution. I really appreciate your efforts.

**Pull request merged**